helpers/lagoonenvclean.sh

```diff
@@ -0,0 +1,182 @@
+#
+# Description: Run a Lagoon old environments cleanup based on closed GitHub's PRs
+# Usage: ./lagoonenvclean customer number_of_months yes|no 
+#
+# Example: ./lagoonenvclean rasch 3 yes 
+# Above command, will revome all environments, older than 3 months, in all customer's projects, related to closed PRs on GitHub.
+
+
+# Script accepts 3 arguments (1 mandatory):
+#
+# CUSTOMER to query
+# MONTHS to calculate the time slot
+# DRYRUN to execute or not the deletion of the environments
+
+
+CUSTOMER="$1"
+MONTHS="${2:-2}"
+DRYRUN="${3:-false}"
+
+
+# CLEANDATE: environments older than that date will be remove if related to a closed PR
+# ENDPOINT: Lagoon API endpoint to query
+# GHENDPOINT: GitHub API endpoint
+# AMZBEARERTKN: Lagoon API token
+# GHUTKN: GitHub API Authentication token
+
+CLEANDATE=$(date +%Y%m%d -d "-$MONTHS months")
+ENDPOINT="https://api-lagoon-master.lagoon.ch.amazee.io/graphql"
+GHENDPOINT="https://api.github.com"
+AMZTKN=""
+AMZBEARERTKN="Authorization: bearer $AMZTKN"
+GHTKN=""
+GHBEARERTKN="Authorization: bearer $GHTKN"
+
+# Some basic Lagoon GraphQL queries:
+#
+# QLPROJECTQUERY: GraphQL query to retrieve all projects belonging to a customer
+# PROJECTQUERYi: one-line and escaped query to retrieve projects of a customer
+# QLENVSQUERY: GraphQL query to retrieve, for each project, all the environments
+# QLENVQUERY: GraphQL query to retrieve information on a single environment per project
+# QLENVDELETE: GraphQL mutation to delete the environment
+
+QLPROJECTQUERY="query allcustomerproject { customerByName (name: \"$CUSTOMER\") { projects { name } }}"
+PROJECTQUERY=$(echo $QLPROJECTQUERY | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+
+QLENVSQUERY="query envbyproject { projectByName (name: \"PROJECT\") { gitUrl, id, environments { name, openshiftProjectName } }}"
+
+QLENVQUERY="query envbyname { environmentByName (name: NAME, project: ID) { updated }}"
+
+QLENVDELETE="mutation delenv { deleteEnvironment(input: {name: ENV, project: PROJECT, execute: BOOLEAN})}"
+
+# Function to retrieve customer's projects
+allproject_query() {
+	PROJECTS=`curl -s -k -X POST -H 'Content-Type: application/json' -H "$AMZBEARERTKN" $ENDPOINT -d "{\"query\": \"$PROJECTQUERY\"}" | jq -r '.[].customerByName.projects[].name'`
+}
+
+#Function to retrieve environments per project
+allenvironment_query() {
+	echo "Lagoon $1 project"
+
+	# Substitute variables and escape characters in the query
+	ENVQUERY=$(echo $QLENVSQUERY | sed "s/PROJECT/$1/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+
+	# Run the query
+	RESULT=`curl -s -k -X POST -H 'Content-Type: application/json' -H "$AMZBEARERTKN" $ENDPOINT -d "{\"query\": \"$ENVQUERY\"}"`
+
+	# From the results, take only environemnts called with a "pr-" to identify Pull Requests
+	ENVS=`echo $RESULT | jq -r '.[].projectByName.environments[].name'|grep -E -i "^(pr-)"`
```

Comment:
there is an actual field `deployType` that tells you if the environment is a PR or a Branch deployment, let's use that one instead of relying on the name (which could be different in the future)

Reply:
Perfect thanks! I changed the logic and I'm using that